### DISCUSSION TITLE
Perform preemptive operations at startup

### DISCUSSION THREAD

**Discussion by sibylantonioforanna** [quartznet/quartznet#1495] [Q&A] [answered]
Hi,
I'd like to perform some operation as soon as QuartzServer starts, for example I'd like to authenticate to some service and shutdown QuartzServer if I can't do that.
This may be useful expecially when QuartzServer is installed as a service so if some needed resource can't be used, the service is stopped and the system administrator can see immediately something is wrong.

Right now an external dll is called only when a job is fired and that may happen in a not so near future, so I'd like to specify some code that is run as soon as quartzServer ends its initialization and before the triggers are fired.

An alternative I've considered is to create a job that starts every n-seconds and pause it after it's processed, but I don't know if it's a viable solution, plus it needs some manual intervention to create this job.

If there's already such a feature can you please point me to some resource/web page where I can find it? I couldn't find it myself.

Thanks in advance.

**Comment by lahma** [comment, marked as answer]
Just copy the quartz server logic to your own console project and validate state before calling scheduler.Start().

**Reply by sibylantonioforanna** [reply]
Sorry, it seems I forgot to add some better context.
My code is inside a dll where a class implements an IJob interface.
A standalone application/GUI registers the jobs, while Quartz server fetches them and starts them by loading the dll.
For what I understand in this scenario the dll is loaded only when a job is fired and the **Execute** method is called.
In fact I don't have a console project, I just use QuartzServer.exe

**Reply by lahma** [reply]
You need to build custom server exe if you want to run custom logic before start.
